### DISCUSSION TITLE
Regarding Error Handling

### DISCUSSION THREAD

**Discussion by Stranger1586** [remix-run/remix#4889] [Q&A] [answered]
Hello there, I was wondering what the remix way of handling errors withing a react component that execute calls to a remix loader is.
Suppose I have this component in `components/NotificationManager.tsx`:

```jsx
export default function NotificationManager() {
  const { data: notificationData, loading } = //fetch data from a remix loader
       return (...)
});
```

And than in my loader `routes/home.ts`:

```jsx
import NotificationManager from 'components/NotificationManager.tsx';
export async function loader(args) {
  return json({...})
}
export default function HomeRoute() {
  return (
....
   <NotificationManager/>
...
  );
}
```

If an error occurs in the remix loader that I am calling from my `NotificationManager` component, I want to be able to handle said error within the component itself, not the route.
Is there some type remix error boundary function that can be used at the component level?

**Comment by machour** [comment, marked as answer]
You can make sure your loader don't throw an error, by try {} catching {} everything in it, and return a different data structure in case of an error.

Then in your component, check the data, and either display your happy path, or an error message.

**Reply by Stranger1586** [reply]
That would be one way to do it, I suppose.
I was wondering if there was some remix mechanism to deal with this particular use case, guess there isnt.
Thank your for your swift response, sir.
Have a good day.
